pkg/p2p/testbed/kadcast_test.go

```diff
@@ -0,0 +1,114 @@
+// This Source Code Form is subject to the terms of the MIT License.
+// If a copy of the MIT License was not distributed with this
+// file, you can obtain one at https://opensource.org/licenses/MIT.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+//go:build testbed
+// +build testbed
+
+package testbed
+
+import (
+	"bytes"
+	"context"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+
+	crypto "github.com/dusk-network/dusk-crypto/hash"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	clusterSize      = os.Getenv("CLUSTER_SIZE")
+	dummyPayloadSize = os.Getenv("MSG_SIZE")
+	ruskExecutable   = os.Getenv("RUSK_PATH")
+
+	NetworkAddr    = "127.0.0.1"
+	bootstrapNodes = []string{NetworkAddr + ":20000", NetworkAddr + ":20001"}
+
+	baseKadcastPort = 20000
+	baseGRPCPort    = 9000
+)
+
+func bootstrapCluster(ctx context.Context, t *testing.T) []*testNode {
+	logrus.WithField("cluster_size", clusterSize).
+		WithField("rusk", ruskExecutable).
+		WithField("message_size", dummyPayloadSize).
+		Info("Bootstrap cluster")
+
+	size, err := strconv.Atoi(clusterSize)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	cluster := make([]*testNode, 0)
+
+	for i := 0; i < size; i++ {
+		n, err := newNode(ruskExecutable, bootstrapNodes, baseKadcastPort+i, baseGRPCPort+i)
+		if err != nil {
+			t.Error(err)
+		}
+
+		switch i {
+		case 1:
+			// Allow bootstrappers to start up
+			time.Sleep(5 * time.Second)
```

Comment:
Using delays to account for startup is ok for tests but can be unnecessarily slow and occasionally too short, a signal is better. A super simple method would be to add a ready signal in the form of an empty file in a common-accessible area between the apps (can even be in tmp), and change this loop to pause a few hundred milliseconds and then check for the existence of the file, and repeat until a timeout.

Reply:
A possible signal mechanism could be the lock file of databases used by Rusk, if this can be easily located and checked by dusk-blockchain

Reply:
No signal mechanism is available for Kadcast bootstrapping and this is simply a test.